test/integration/connection-monitoring-and-pooling/rtt_pinger.test.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from 'chai';
+import * as semver from 'semver';
+import * as sinon from 'sinon';
+
+import { type Connection, type MongoClient, type RTTPinger } from '../../mongodb';
+import { sleep } from '../../tools/utils';
+
+/**
+ * RTTPinger creation depends on getting a response to the monitor's initial hello
+ * and that hello containing a topologyVersion.
+ * Subsequently the rttPinger creates its connection asynchronously
+ *
+ * I just went with a sleepy loop, until we have what we need, One could also use SDAM events in a clever way perhaps?
+ */
+async function getRTTPingers(client: MongoClient) {
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const rttPingers = Array.from(client.topology?.s.servers.values() ?? [], s => {
+      if (s.monitor?.rttPinger?.connection != null) return s.monitor?.rttPinger;
+      else null;
+    }).filter(rtt => rtt != null);
+
+    if (rttPingers.length !== 0) {
+      return rttPingers as (Omit<RTTPinger, 'connection'> & { connection: Connection })[];
+    }
+
+    await sleep(5);
+  }
+}
+
+describe('class RTTPinger', () => {
+  afterEach(() => sinon.restore());
+
+  beforeEach(async function () {
+    if (this.configuration.isLoadBalanced) {
+      if (this.currentTest)
+        this.currentTest.skipReason = 'No monitoring in LB mode, test not relevant';
+      return this.skip();
+    }
+    if (semver.gte('4.4.0', this.configuration.version)) {
+      if (this.currentTest)
+        this.currentTest.skipReason =
+          'Test requires streaming monitoring, needs to be on MongoDB 4.4+';
+      return this.skip();
+    }
+  });
+
+  context('when serverApi is enabled', () => {
```

Comment:
what about when it's not?

Reply:
Agreed so the last case we're missing here is "serverApi disabled, server supports hello". added!